qt_ui/windows/AirWingConfigurationDialog.py

```diff
@@ -79,25 +82,35 @@ class SquadronBaseSelector(QComboBox):
     def __init__(
         self,
         bases: Iterable[ControlPoint],
-        squadron: Squadron,
+        aircraft_type: Optional[AircraftType],
+        current_value: str,
```

Comment:
`Optional` so `"Select aircraft type first"` isn't part of the API? It should probably be `Optional[ControlPoint]` so changing the display format doesn't change the default selection.

Reply:
Sorry i dont really understand what you mean. You say you want to get the "Select aircraft type first" thing out of the SquadronBaseSelector class? I did this quirks here because it is only used in this specific context and does not make any sense to use when no aircraft is given for it. Could make it better of course but not sure if its worh the effort :D

Reply:
This currently accepts effectively `Union[ControlPointName, Literal["Select aircraft type first"]]`. If the display format of `SquadronBaseSelector` items changes (say we decide to include the CP type in the drop down, or we change to `"Select aircraft type"` or whatever), the call sites need to change too.

It could instead take `Optional[ControlPoint]` so that the callers don't need to know the display format. This is how similar widgets handle this behavior. https://github.com/dcs-liberation/dcs_liberation/blob/develop/qt_ui/windows/mission/flight/SquadronSelector.py for an example.

I agree it's a minor thing so if you feel strongly about it I won't be a stickler about it :)

Reply:
> his is how similar widgets handle this behavior. https://github.com/dcs-liberation/dcs_liberation/blob/develop/qt_ui/windows/mission/flight/SquadronSelector.py for an example.

thanks for the clarification. Will have a look into that one and adopt it accordingly :)